### ISSUE TITLE
Java14 Full Records Support Check Validation: IllegalTypeCheck

### ISSUE THREAD

**Issue by nmancus1** [checkstyle/checkstyle#8468]
Check documentation: https://checkstyle.sourceforge.io/config_coding.html#IllegalType

From check documentation:
_Checks that particular classes or interfaces are never used._

```
➜  full-record-grammar /usr/lib/jvm/java-14-openjdk/bin/javac --enable-preview --source 14 TestClass.java
Note: TestClass.java uses preview language features.
Note: Recompile with -Xlint:preview for details.
➜  full-record-grammar cat config.xml    
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
      <module name="IllegalType"/>
  </module>
</module>
➜  full-record-grammar cat TestClass.java
import java.util.HashMap;
import java.util.LinkedHashMap;
import java.util.Queue;
import java.util.TreeMap;

public class TestClass {
    public record MyRecord1() {
        static LinkedHashMap<Integer, String> lhmap =
                new LinkedHashMap<Integer, String>(); // violation
        static TreeMap<Integer, String> treemap = new TreeMap<Integer, String>(); // violation
        static HashMap<String, String> hmap; // violation
        static Queue<Integer> intqueue; // OK

       // not allowed in records: java.lang.IllegalArgumentException illegalex;
       // not allowed in records: java.util.TreeSet treeset;
    }

}
➜  full-record-grammar java $RUN_LOCALE -jar ~/IdeaProjects/checkstyle/target/checkstyle-8.35-SNAPSHOT-all.jar -c config.xml TestClass.java
Starting audit...
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:8:16: Usage of type 'LinkedHashMap' is not allowed. [IllegalType]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:10:16: Usage of type 'TreeMap' is not allowed. [IllegalType]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:11:16: Usage of type 'HashMap' is not allowed. [IllegalType]
Audit done.
Checkstyle ends with 3 errors.
```
This check works as intended, I think we can close this issue.

**Comment by pbludov** [comment]
A forbidden type may come from `implements` or generics:
```xml
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
    "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="IllegalType">
            <property name="illegalClassNames" value="Cloneable"/>
        </module>
    </module>
</module>
```

```java
class TestClass implements Cloneable { // violation
}

record TestRecord() implements Cloneable { // expected violation
}
```

```
Starting audit...
[ERROR] /home/pbludov/src/tmp/TestClass.java:1:28: Usage of type 'Cloneable' is not allowed. [IllegalType]
Audit done.
```
